src/flows/CostCalculator/CostCalculatorForm.tsx

```diff
@@ -54,9 +60,23 @@ export function CostCalculatorForm({
     } else {
       if (costCalculatorResults?.data) {
         await onSuccess?.(costCalculatorResults?.data);
+        // Reset the form completely
         if (shouldResetForm) {
           costCalculatorBag?.resetForm();
           form.reset();
+          return;
+        }
+
+        // Reset only the form fields
+        if (shouldResetFormFields) {
+          const currentValues = form.getValues();
+          const resetValues = { ...currentValues };
+          shouldResetFormFields.forEach((field) => {
+            resetValues[field] = '';
+          });
+
+          costCalculatorBag?.resetForm();
+          form.reset(resetValues);
```

Comment:
According to the [docs](https://react-hook-form.com/docs/useform/reset), check `Submit with Reset` section, the reset should be done inside of a useEffect.